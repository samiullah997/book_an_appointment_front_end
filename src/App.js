import {
  BrowserRouter, Outlet, Route, Routes,
} from 'react-router-dom';
import LogIn from './components/authentication/LogIn';
import SignUp from './components/authentication/SignUp';
import AddBike from './components/dashboard/AddBike';
import DeleteBikes from './components/dashboard/DeleteBikes';
import Details from './components/dashboard/Details';
import Home from './components/dashboard/Home';
import Reservation from './components/dashboard/Reservation';
import ReserveBike from './components/dashboard/ReserveBike';
import ReserveList from './components/dashboard/ReserveList';
import SideBar from './components/sidebar/sideBar';

const SidebarLayout = () => (
  <>
    <SideBar />
    <Outlet />
  </>
);
function App() {
  return (
    <div className="flex w-screen justify-start h-screen items-center">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<LogIn />} />
          <Route path="/register" element={<SignUp />} />
          <Route path="/user/Reservation" element={<Reservation />} />
          <Route path="/user/ReserveBike" element={<ReserveBike />} />
          <Route element={<SidebarLayout />}>
            <Route path="/user/dashboard" element={<Home />} />
            <Route path="/user/AddBike" element={<AddBike />} />
            <Route path="/user/DeleteBikes" element={<DeleteBikes />} />
            <Route path="/user/Details" element={<Details />} />
            <Route path="/user/MyReservations" element={<ReserveList />} />
          </Route>
        </Routes>
      </BrowserRouter>
    </div>
  );
}
export default App;
