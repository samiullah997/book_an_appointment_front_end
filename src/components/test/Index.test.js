import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../../redux/Store';
import App from '../../App';

const MockIndex = () => (
  <Provider store={store}>
    <App />
  </Provider>
);

it('should match snapshot of the Index component', () => {
  const tree = renderer.create(<MockIndex />).toJSON();
  expect(tree).toMatchSnapshot();
});
