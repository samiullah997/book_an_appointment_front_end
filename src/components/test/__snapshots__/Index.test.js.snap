// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot of the Index component 1`] = `
<div
  className="flex w-screen justify-start h-screen items-center"
>
  <div
    className="flex h-screen w-full items-center justify-center bg-gray-900 bg-cover bg-no-repeat bg-image"
  >
    <div
      className="rounded-xl bg-white bg-opacity-50 px-16 py-10 shadow-lg backdrop-blur-md max-sm:px-8"
    >
      <div
        className="text-white"
      >
        <div
          className="mb-8 flex flex-col items-center"
        >
          <img
            alt=""
            src="./favicon.png"
            width="150"
          />
          <h1
            className="mb-2 text-2xl"
          >
            Buy Bikes
          </h1>
          <span
            className="text-white"
          >
            Enter Login Details
          </span>
          <span
            className="text-xl text-red-700 font-bold mt-4"
          />
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-4 text-lg"
          >
            <input
              className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
              name="name"
              onChange={[Function]}
              placeholder="id@email.com"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="mb-4 text-lg"
          >
            <input
              className="rounded-3xl border-none bg-black bg-opacity-50 px-6 py-2 text-center text-inherit placeholder-slate-200 shadow-lg outline-none backdrop-blur-md"
              name="name"
              onChange={[Function]}
              placeholder="*********"
              required={true}
              type="Password"
              value=""
            />
          </div>
          <div
            className="mt-8 flex justify-center text-lg text-black"
          >
            <button
              className="rounded-3xl bg-black bg-opacity-50 px-10 py-2 text-white shadow-xl backdrop-blur-md transition-colors duration-300 hover:bg-yellow-600"
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
        <p
          className="mt-8 text-xs font-light text-center text-gray-700"
        >
           
          Do not have an account?
          <a
            className="font-medium text-purple-600 hover:underline"
            href="/signup"
          >
            Sign up
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
